### PR TITLE
Add Angel Sanchez Lab3 solution

### PR DESCRIPTION
**Pull request by Angeloyo**

### CODE REVIEW

**Review by bardhprenkaj — APPROVED** [review]
Exactly what one should do when doing hyperparameter optimization: i.e., perform the optimization on the validation set. I would have also wanted you to calculate the performance on the test set after the optimization, but it's fine.
